src/Hamiltonians/ExtendedHubbardMom1D.jl

```diff
@@ -0,0 +1,128 @@
+
+LOStructure(::Type{<:ExtendedHubbardMom1D{<:Real}}) = IsHermitian()
+
+Base.getproperty(h::ExtendedHubbardMom1D, s::Symbol) = getproperty(h, Val(s))
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:ks}) = getfield(h, :ks)
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:kes}) = getfield(h, :kes)
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:address}) = getfield(h, :address)
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,U}, ::Val{:u}) where {U} = U
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,V}, ::Val{:v}) where {V} = V
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,<:Any,T}, ::Val{:t}) where {T} = T
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,<:Any,<:Any,BOUNDARY_CONDITION}, 
+    ::Val{:boundary_condition}) where {BOUNDARY_CONDITION} = BOUNDARY_CONDITION
+
+ks(h::ExtendedHubbardMom1D) = getfield(h, :ks)
+
+# standard interface function
+function num_offdiagonals(ham::ExtendedHubbardMom1D, address::SingleComponentFockAddress)
+    singlies, doublies = num_singly_doubly_occupied_sites(address)
+    return num_offdiagonals(ham, address, singlies, doublies)
+end
+
+# 4-argument version
+@inline function num_offdiagonals(ham::ExtendedHubbardMom1D, ::SingleComponentFockAddress, singlies, doublies)
+    M = num_modes(ham)
+    return singlies * (singlies - 1) * (M - 2) + doublies * (M - 1)
+end
+
+@inline function momentum_transfer_diagonal(
+    h::ExtendedHubbardMom1D{<:Any,M,<:BoseFS}, map
+) where {M}
+    return h.u / 2M *momentum_transfer_diagonal(map) 
+    + h.v/ M * dot(cos.(h.ks), map) * extended_momentum_transfer_diagonal(map,M) 
+end
+
+@inline function momentum_transfer_diagonal(
+    h::ExtendedHubbardMom1D{<:Any,M,<:FermiFS}, map
+) where {M}
+    return (h.v/ M) * extended_momentum_transfer_diagonal(map, M)
+end
+
+@inline function diagonal_element(h::ExtendedHubbardMom1D, address::SingleComponentFockAddress)
+    map = OccupiedModeMap(address)
+    return dot(h.kes, map) + momentum_transfer_diagonal(h, map)
+end
+
+@inline function get_offdiagonal(
+    ham::ExtendedHubbardMom1D{<:Any,M,A}, address::A, chosen, map=OccupiedModeMap(address)
+) where {M,A<:BoseFS}
+    address, onproduct,_,_,q = momentum_transfer_excitation(address, chosen, map)
+    return address, ham.u/(2*M)*onproduct + ((ham.v * cos((q)*((2*π)/M)))/(M))*onproduct
+end
+
+@inline function get_offdiagonal(
+    ham::ExtendedHubbardMom1D{<:Any,M,A}, address::A, chosen, map=OccupiedModeMap(address)
+) where {M,A<:FermiFS}
+    address, onproduct,_,_,q = momentum_transfer_excitation(address, chosen, map)
+    return address, -((ham.v * cos((q)*((2*π)/M)))/(M))*onproduct
+end
+
+function offdiagonals(h::ExtendedHubbardMom1D, a::SingleComponentFockAddress)
+    map = OccupiedModeMap(a)
+    singlies = length(map)
+    doublies = count(i -> i.occnum ≥ 2, map)
+    num = num_offdiagonals(h, a, singlies, doublies)
+    return OffdiagonalsBoseMom1D(h, a, num, map)
+end
```

Comment:
Hmm, this looks a bit like unnecessary duplication as it is identical to the code for `HubbardMom1D` and could be eliminated by changing the dispatch rules for the latter. Maybe it is still valuable (for readability) to have the code here?

We should also think about whether it makes sense to only have `HubbardMom1D`, where the extended version is chosen if `v != 0`.